File: HUI2/src/services/WishlistService.js
import axios from "axios";

const API = `${import.meta.env.VITE_API_URL}/wishlist`;
export const getWishlist = (email) => axios.get(`${API}/${email}`);

export const addToWishlist = (item) => axios.post(API, item);

export const removeFromWishlist = (pid, email) =>
    axios.delete(`${API}/${pid}/${email}`);

export const checkWishlist = (pid, email) =>
    axios.get(`${API}/check/${pid}/${email}`);

File: HerbalLife/src/main/java/com/example/HerbalLife/service/WishlistService.java
package com.example.HerbalLife.service;

import java.util.List;

import com.example.HerbalLife.entity.Wishlist;

public interface WishlistService {

    Wishlist addToWishlist(Wishlist wishlist);

    List<Wishlist> getWishlist(String email);

    void removeFromWishlist(int pid, String email);

    boolean isInWishlist(int pid, String email);

}
